Doc/howto/annotations.rst

```diff
@@ -57,6 +57,12 @@ Accessing The Annotations Dict Of An Object In Python 3.10 And Newer
   newer is to call :func:`getattr` with three arguments,
   for example ``getattr(o, '__annotations__', None)``.
 
+  Before Python 3.10, accessing ``__annotations__`` on a class that
+  defines no annotations but that has a parent class with
+  annotations would return the parent's ``__annotations__``.
+  In Python 3.10 and newer, the child class's annotations
+  will be an empty dict instead.
```

Comment:
As mentioned in https://github.com/python/cpython/issues/99535#issuecomment-1355612259, there is _one_ edge case where this isn't true: `TypedDict`s still inherit a parent class's `__annotations__`, even in Python 3.10. I'm not sure whether it's important to mention that or not -- @hauntsaninja, do you have any thoughts on that?

```pycon
Python 3.10.9 (tags/v3.10.9:1dd9be6, Dec  6 2022, 20:01:21) [MSC v.1934 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> from typing import TypedDict
>>> class Foo(TypedDict):
...     x: int
...
>>> class Bar(Foo): ...
...
>>> Bar.__annotations__
{'x': <class 'int'>}
```

Reply:
I think it's fine not to mention over here:
1) The special case is more that TypedDicts don't have a runtime subclassing relationship, rather than anything to do with `__annotations__`. In other words, if you know how TypedDict subclassing works at runtime (which is presumably documented somewhere), you can reasonable expect this `__annotations__` behaviour. So adding more words here might be confusing; the statement about the language remains true
2) My understanding of "HOWTO"s is that they're allowed to skip some of the gross details